qhub/cli/_init.py

```diff
@@ -0,0 +1,496 @@
+
+    # specific context needed when `check_project_name` is called
+    ctx.params["cloud_provider"] = inputs.cloud_provider
+
+    # PROJECT NAME
+    rich.print(
+        (
+            "\n 🪴  Next, give your Nebari instance a project name. This name is what your Kubernetes cluster will be referred to as.\n\n"
+        )
+    )
+    inputs.project_name = questionary.text(
+        "What project name would you like to use?",
+        qmark=qmark,
+        validate=lambda text: True if len(text) > 0 else "Please enter a value",
+    ).ask()
+
+    if not disable_checks:
+        check_project_name(ctx, inputs.project_name)
+
+    # DOMAIN NAME
+    rich.print(
+        (
+            "\n 🪴  Great! Now it's time to decide on the domain name (i.e the URL) that Nebari will be accessible from. "
+            "This should be domain that you own.\n\n"
+        )
+    )
+    inputs.domain_name = questionary.text(
+        "What domain name would you like to use?",
+        qmark=qmark,
+        validate=lambda text: True if len(text) > 0 else "Please enter a value",
+    ).ask()
+
+    # AUTH PROVIDER
+    rich.print(
+        (
+            "\n 🪴  Nebari comes with [green]Keycloak[/green], an open-source identity and access management tool. This is how users, groups and roles "
+            "are managed on the platform. To connect Keycloak with an identity provider, you can select one now.\n\n"
+            "\n\t❕ [purple]password[/purple] is the default option and is not connected to any external identity provider.\n"
+        )
+    )
+    inputs.auth_provider = questionary.select(
+        "What authentication provider would you like?",
+        choices=enum_to_list(AuthenticationEnum),
+        qmark=qmark,
+    ).ask()
+
+    if not disable_checks:
+        check_auth_provider_creds(ctx, auth_provider=inputs.auth_provider)
+
+    if inputs.auth_provider.lower() == AuthenticationEnum.auth0.value.lower():
+        inputs.auth_auto_provision = questionary.confirm(
+            "Would you like us to auto provision the Auth0 Machine-to-Machine app?",
+            default=False,
+            qmark=qmark,
+        ).ask()
+
+    elif inputs.auth_provider.lower() == AuthenticationEnum.github.value.lower():
+        rich.print(
+            (
+                ":warning: If you haven't done so already, please ensure the following:\n"
+                f"The `Homepage URL` is set to: [light_green]https://{inputs.domain_name}[/light_green]\n"
+                f"The `Authorization callback URL` is set to: [light_green]https://{inputs.domain_name}/auth/realms/qhub/broker/github/endpoint[/light_green]\n\n"
+            )
+        )
+
+    # GITOPS - REPOSITORY, CICD
+    rich.print(
+        (
+            "\n 🪴  This next section is [italic]optional[/italic] but recommended. If you wish to adopt a GitOps approach to managing this platform, "
+            "we will walk you through a set of questions to get that setup.\n\n"
```

Comment:
It might be worth sharing what GitOps would entail, something like:

> With this setup, Nebari will use GitHub Actions workflows (or GitLab equivalent) to automatically handle the future deployments of your infrastructure.

(The above sentence is taken from: https://docs.qhub.dev/en/stable/source/installation/usage.html#gitops)

Reply:
That's a good call!